### PR TITLE
refactor: extract config to separate module

### PR DESCRIPTION
**Pull request by MunifTanjim**

### CODE REVIEW

**Comment by zbirenbaum** [comment]
I like the idea here but I want to test a few things before it's merged. Can you let me know when it's in a state that I should review?

**Comment by MunifTanjim** [comment]
> I like the idea here but I want to test a few things before it's merged. Can you let me know when it's in a state that I should review?

It's already done. I was just trying it out a couple days to confirm nothing is breaking, that's why it was in draft state. You can review it now if you want.

**Review by zbirenbaum — APPROVED** [review]
Looks great!

**Pull request merged**
